.github/workflows/teamcity-nightly-sweeper.yaml

```diff
@@ -0,0 +1,102 @@
+# one branch that matches the filter (+:UTC-*,-:UTC-nightly-tests*)
+
+on:
+    workflow_call:
+    workflow_dispatch:
+      inputs:
+        dayThreshold:
+          default: '3'
+    schedule:
+        - cron: '0 14 * * *' # UTC 2PM (-7)-> 7AM PST
+
+jobs:
+  rename-TC-nightly-branch:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/github-script@e69ef5462fd455e02edcaf4dd7708eda96b9eda0 # v7.0.0
+        with:
+          retries: 3
+          retry-exempt-status-codes: 400, 401, 403, 404, 422
+          script: |
+                  let dateToday = new Date().toJSON().slice(0, 10);
+                  const oldBranchName = "UTC-" + dateToday;
+                  const newBranchName = "UTC-nightly-tests-" + dateToday;
+                  
+                  try{
+                  await github.rest.repos.renameBranch({
+                    owner: context.repo.owner,
+                    repo: context.repo.repo,
+                    branch: oldBranchName,
+                    new_name: newBranchName, 
+                  })
+                  } catch (error){
+                    core.setFailed(error + "- Failed to rename branch; branch with name " + oldBranchName + " doesn\'t exist")
+                  }
+                  console.log("Renamed branch " + oldBranchName + " to " + newBranchName)
+
+  sweeping-outdated-branches:
+    needs: rename-TC-nightly-branch
+    runs-on: ubuntu-latest
+    steps: 
+      - uses: actions/github-script@e69ef5462fd455e02edcaf4dd7708eda96b9eda0 # v7.0.0
+        env:
+          DAYS_THRESHOLD: ${{ inputs.dayThreshold || '3'}} # this allows the default value to be 3 when triggered on schedule
+        with:
+          retries: 3
+          retry-exempt-status-codes: 400, 401, 403, 404, 422
+          script: |
+            const { DAYS_THRESHOLD } = process.env
+            console.log(`Removing nightly-test branches not made in the last ${DAYS_THRESHOLD} days`)
+        
+            function dateDifference(dateToday, branchDate){
+                dateToday = new Date(dateToday)
+                branchDate = new Date(branchDate)
+                return (dateToday - branchDate) / 86_400_000 // calculates the difference in days based on milliseconds
+            }
+            
+            async function branchSweeper(daysThreshold){
+                let dateToday = new Date().toJSON().slice(0, 10);
+                console.log("Today\'s date: ",dateToday);
+                // grab the list of branches then iterate through the list checking for the difference in days
+                const branchList = await github.rest.repos.listBranches({
+                    owner: context.repo.owner,
+                    repo: context.repo.repo,
+                    protected: false
+                  })
+
+                const filteredBranches = branchList.data.filter( (branch) => {
+                  const branchDate = /^UTC-nightly-\d{4}-\d{2}-\d{2}$/g.exec(branch.name)
```

Comment:
This doesn't match the `UTC-nightly-tests-YYYY-MM-DD` format used in in the `rename-TC-nightly-branch` job - please update so we can find branches to sweep!